addons/project/wizard/project_share_wizard.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, Command, fields, models, _
+from odoo.exceptions import ValidationError, UserError
+
+from ..models.project_sharing_access import PROJECT_SHARING_ACCESS_MODE
+
+
+class ProjectShareWizard(models.TransientModel):
+    _name = 'project.share.wizard'
+    _inherit = 'portal.share'
+    _description = 'Project Share'
+
+    @api.model
+    def default_get(self, fields_list):
+        result = super().default_get(fields_list)
+        if result['res_model'] and result['res_id']:
+            project = self.env[result['res_model']].browse(result['res_id'])
+            result.update(
+                allow_project_sharing=project.allow_project_sharing,
+                access_ids=[
+                    Command.create(
+                        {'user_id': access.user_id, 'access_mode': access.access_mode}
+                    ) for access in project.project_sharing_access_ids
+                ],
+            )
+        return result
+
+    partner_ids = fields.Many2many(compute="_compute_partner_ids", readonly=False, store=True, required=False)
+    access_ids = fields.One2many('project.share.access.wizard', 'project_share_id')
```

Comment:
`user_ids` or `line_ids`, for a more readable name, wrt to the "child" model name suggestion below?